pkg/apis/monitoring/v1/types.go

```diff
@@ -150,6 +150,109 @@ func (pc *ProxyConfig) Validate() error {
 	return nil
 }
 
+type SafeHTTPHeader struct {
+	// Header values.
+	//
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:items:MinLength=1
+	// +listType=set
+	// +optional
+	Values []string `json:"values,omitempty"`
+
+	// Headers values. Hidden in configuration page.
+	//
+	// +kubebuilder:validation:MinItems=1
+	// +listType=set
+	// +optional
+	SecretRefs []v1.SecretKeySelector `json:"secretRefs,omitempty"`
+}
+
+// Validate semantically validates the given SafeHttpHeader.
+func (c *SafeHTTPHeader) Validate() error {
+	if c == nil {
+		return nil
+	}
+
+	if reflect.ValueOf(c).IsZero() {
+		return nil
+	}
+
+	for _, v := range c.SecretRefs {
+		if v == (v1.SecretKeySelector{}) {
+			return errors.New("Secrets selector must be defined")
+		}
+	}
+
+	return nil
+}
+
+type HTTPHeader struct {
+	// Name of the referent.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+
+	SafeHTTPHeader `json:",inline"`
+
+	// Files to read header values from.
+	//
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:items:MinLength=1
+	// +listType=set
+	// +optional
+	Files []string `json:"files,omitempty"`
```

Comment:
Sorry if my previous comment wasn't clear but I'd rather focus on implement secret key references for now and leave other fields to a latter PR (allowing file paths looks sensible but I'm more skeptical on allowing bare values). 
```suggestion
	// References to secret keys holding the header values.
	//
	// +kubebuilder:validation:MinItems=1
	// +optional
	SecretRefs []v1.SecretKeySelector `json:"secretRefs,omitempty"`
```

Reply:
Thanks @simonpasquier 
I got it
Change struct done. 
Other fields will be added in subsequent PRs.